project/src/main/kotlin/org/ionproject/core/accessControl/BootTokenGeneration.kt

```diff
@@ -0,0 +1,59 @@
+package org.ionproject.core.accessControl
+
+import org.ionproject.core.accessControl.TokenGenerator.Companion.buildToken
+import org.ionproject.core.accessControl.TokenGenerator.Companion.encodeBase64
+import org.ionproject.core.accessControl.TokenGenerator.Companion.generateRandomString
+import org.ionproject.core.accessControl.TokenGenerator.Companion.getHash
+import org.ionproject.core.accessControl.pap.sql.AuthRepo
+import org.ionproject.core.common.interceptors.LoggerInterceptor
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+private val logger = LoggerFactory.getLogger(LoggerInterceptor::class.java)
+
+@Component
+class BootTokenGeneration(private val authRepo: AuthRepo) {
```

Comment:
I thought this was going to happen at deployment time and not at boot time. Notice that the app can restart multiple times for the same deployment. The tokens should not change on that case.